### PR TITLE
Various fixes and improvements in Regularization of linear regression model

### PR DESCRIPTION
**Pull request by ArturoAmorQ**
Follows #698, which I merged without a proper final pass.

Hopefully this PR improves some final details.

### CODE REVIEW

File: python_scripts/linear_models_regularization.py [hunk 1]
```diff
@@ -98,14 +98,15 @@
 # features with a `PolynomialFeatures` transformer. For instance, one does not
 # expect features such as `PoolArea * YrSold` to be very predictive.
 #
-# We can create a dataframe to check the weights of the model: the columns
-# contain the name of the features whereas the rows store the coefficients values
-# of each model during the cross-validation.
+# To analyze the weights of the model, we can create a dataframe. The columns of
+# the dataframe will contain the feature names, while the rows will store the
```
Comment (glemaitre) [inline python_scripts/linear_models_regularization.py:102]:
```suggestion
# the dataframe contains the feature names, while the rows will store the
```

File: python_scripts/linear_models_regularization.py [hunk 2]
```diff
@@ -315,19 +316,20 @@
 weights_ridge_scaled_data = pd.DataFrame(coefs, columns=feature_names)
 
 # %%
-fig, ax = plt.subplots(figsize=(8, 12))
+fig, ax = plt.subplots(figsize=(8, 10))
 weights_ridge_scaled_data.plot.box(color=color, vert=False, ax=ax)
 _ = ax.set(
     title="Ridge weights with data scaling",
-    xscale="symlog",
+    # xscale="symlog",
```
Comment (glemaitre) [inline python_scripts/linear_models_regularization.py:323]: I did not look at the rendering yet but this should probably go away.

**Comment by glemaitre** [comment]
There are typos in the following paragraph:

```
# Notice that we use a (symmetric) log scale for the bar plot. Observe that some
# coefficents are extremely large while others are extremely small. Furthermore,
# the coefficient values can be very unstable accross cross-validation folds.
```

coefficents -> coefficients
accross -> across

File: python_scripts/linear_models_regularization.py (hunk 2 above)
Comment (glemaitre) [reply python_scripts/linear_models_regularization.py:323]: We should not use the symlog for the other weights also.

**Review by glemaitre — COMMENTED** [review]
Otherwise, LGTM.

**Review by glemaitre — APPROVED** [review]

**Review by ogrisel — APPROVED** [review]
LGTM as well!

**Pull request merged**